rfcs/README.md

```diff
@@ -0,0 +1,139 @@
+
+Many changes, including bug small to medium features, fixes, and documentation 
+improvements can be implemented and reviewed via the normal GitHub pull request 
+workflow.
+
+Some changes though are "substantial", and we ask that these be put
+through a bit of a design process and produce a consensus among the relevant
+Kibana team.
+
+The "RFC" (request for comments) process is intended to provide a
+consistent and controlled path for new features to enter the project.
+
+[Active RFC List](https://github.com/elastic/kibana/pulls?q=is%3Aopen+is%3Apr+label%3ARFC)
+
+Kibana is still **actively developing** this process, and it will still change as
+more features are implemented and the community settles on specific approaches
+to feature development.
+
+## Contributor License Agreement (CLA)
+
+In order to accept your pull request, we need you to submit a CLA. You only need
+to do this once, so if you've done this for another Elastic open source
+project, you're good to go.
+
+**[Complete your CLA here.](https://www.elastic.co/contributor-agreement)**
+
+## When to follow this process
+
+You should consider using this process if you intend to make "substantial"
+changes to Kibana or its documentation. Some examples that would benefit
+from an RFC are:
+
+   - A new feature that creates new API surface area, such as a new core
+     service available to plugins.
+   - The removal of features that already shipped as part of a release.
+   - The introduction of new idiomatic usage or conventions, even if they
+     do not include code changes to Kibana itself.
+
+The RFC process is a great opportunity to get more eyeballs on your proposal
+before it becomes a part of a released version of Kibana. Quite often, even
+proposals that seem "obvious" can be significantly improved once a wider
+group of interested people have a chance to weigh in.
+
+The RFC process can also be helpful to encourage discussions about a proposed
+feature as it is being designed, and incorporate important constraints into
+the design while it's easier to change, before the design has been fully
+implemented.
+
+Some changes do not require an RFC:
+
+  - Rephrasing, reorganizing or refactoring
+  - Addition or removal of warnings
+  - Additions that strictly improve objective, numerical quality
+  criteria (speedup, better browser support)
+  - Addition of features that do not impact other Kibana plugins (do not
+  expose any API to other plugins)
+
+## What the process is
+
+In short, to get a major feature added to Kibana Core, one usually first gets
+the RFC merged into the RFC tree as a markdown file. At that point the RFC
+is 'active' and may be implemented with the goal of eventual inclusion
+into Kibana.
+
+* Fork the Kibana repo http://github.com/elastic/kibana
+* Copy `rfcs/0000_template.md` to `rfcs/text/0000_my_feature.md` (where
+'my_feature' is descriptive. Don't assign an RFC number yet).
```

Comment:
When do we assign it a number? Is the number overkill? We'll probably end up referring to RFCs by their name anyway.

Reply:
I slightly prefer using numbers just to make sure what we're referencing is very clear. It's possible that some RFCs end up having similar names. In terms of when to assign the number, I think it'd be fine if we assign the number when we create the PR. It's unlikely we'll have conflicts with the size of our team.

Reply:
How do I find the latest RFC number when I am creating an RFC?